scripts/gitfunctions.sh

```diff
@@ -2,43 +2,196 @@
+      ;;
+    r) recurse=true
+      ;;
+    h|\?)
+      echo -e "Usage: gitf [-r] [-q] [-h|-?] <[--] Git instructions>
+Recursively executes the given Git command on the current and its direct children Git repositories.\n
+\t-h,-?\tShow this help message and exit.
+\t-q\tQuiet mode (default is verbose).
+\t-r\tThe recurse will continue on all children Git repositories.
+\t--\tOptional separator between the options and the instructions which may start with a dash."
+      return
+      ;;
+    esac
+  done
+  shift $((OPTIND-1))
+  [ "$1" = "--" ] && shift
+  local git_args=$@
+
+  function git_command {
+    local dir
+    if [ -e .git ]; then
+      dir=$(basename $PWD)
       echo "[$dir]"
-      (cd "$dir" ; git "$@")
+      [ "$quiet" != true ] && echo "\$> git $git_args"
+      (eval git $git_args) || true
       echo
     fi
-  done
+    for dir in $(ls -d */); do
+      dir=${dir%%/};
+      if [ -e "$dir"/.git ]; then
+        ( cd "$dir" || continue
+          if [ "$recurse" = true ]; then
+              git_command
+          else
+            echo "[$dir]"
+            [ "$quiet" != true ] && echo "\$> git $git_args"
+            (eval git $git_args) || true
+            echo
+          fi
+        )
+      fi
+    done
+  }
+  git_command
 }
 
 gitfa() {
-  git_args=$@
+  # Parse options
+  local OPTIND=1 # getopts reset
+  local quiet=false
+  local opt
+  while getopts "qh?" opt; do
+    case "$opt" in
+    q) quiet=true
+      ;;
+    h|\?)
+      echo -e "Usage: gitfa [-q] [-h|-?] <[--] Git instructions>
+Recursively executes the given Git command on the current, its direct children Git repositories \
+and browse the child Maven modules of \$PARENT_MODULES directories.\n
+\t-h,-?\tShow this help message and exit.
+\t-q\tQuiet mode (default is verbose).
+\t--\tOptional separator between the options and the instructions which may start with a dash.
+\tPARENT_MODULES Environment variable. Defaults to 'addons addons-core' if not set.
+\tLIST_<parent module> Environment variable that can be set to restrict the children modules to a fixed list. \
+For instance: LIST_ADDONS_CORE=\"nuxeo-core-storage-marklogic\" or LIST_ADDONS=\"nuxeo-shell nuxeo-quota\"."
+      return
+      ;;
+    esac
+  done
+  shift $((OPTIND-1))
+  [ "$1" = "--" ] && shift
+  local git_args=$@
+
+  local parent_modules=${PARENT_MODULES:-"addons addons-core"}
   function git_command {
     if [ -e .git ]; then
       echo "[$1]"
-      git $git_args
+      [ "$quiet" != true ] && echo "\$> git $git_args"
+      (eval git $git_args) || true
       echo
     fi
   }
-  _execute_on_modules "git_command"
+  local dir=$(basename $PWD)
+  git_command $dir
+  # Detect if we are at Nuxeo Platform repository's root
+  if [ -d "nuxeo-common" ]; then
+    local dir
+    for dir in $(ls -d */); do
+      ( dir=${dir%%/};
+        cd "$dir" || continue
+        git_command "$dir"
+        if [[ "$parent_modules" =~ (^|[[:space:]])"$dir"($|[[:space:]]) ]]; then
```

Comment:
I'd just do (portable) (note the spaces):

    case " $parent_modules " in
        *" $dir "*) _execute_on_modules "git_command"
            ;;
    esac